File: scpxfer/xfer.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
   string username; 
   string destination;
   int ports;
   string fifol;
   string final_dest;
   cout << "WECOME TO THE SCP FILE XFER CMD PREP PROGRAM\n" <<endl;
   cout << "Give me your destination's username: "; // username input
   cin >> username; // get user input  
   cout << "Your destination's username is: " << username <<endl; 
   cout << "give me your destination (place.ipaddress.org): "; // destination input 
   cin >> destination;
   cout << "your destination is: " << destination << endl;
   cout << "any ports? enter them here (default is 22): "; //port input
   cin >> ports;
   cout << "your port is: " << ports << endl;
   cout << "which file/folder are we moving?: ";
   cin >> fifol;
   cout << "your file is: " << fifol <<endl;
   cout << "where do you want it at the destination?: ";
   cin >> final_dest;
   cout << "your destination directory is: " << final_dest << endl;
   std::string boiler = "scp -r -P";
   cout << "copy and paste the following command into terminal \n";
   cout << "your final command is: "<<boiler<<" "<<ports<<" "<<fifol<<" "<<username<<'@'<<destination<<':'<<final_dest<<endl;


   return 0;
}

File: scpxfer/xfer.py
#!/usr/bin/env python

import os

# artsy title 
print('\n')
print('/'*50,'\n',' '*2,'WELCOME TO THE SCP FILE TRANSFER PROGRAM')
print('/'*50,)
print('/'*25,)
print('/'*10,)

# begin input parts

fi_of_interest = input('\n\nWhat & where is the file? \n(eg. path/to/file.xyz)\nIf needed use command \' $ pwd \' \n\nEnter here --> ')

usern = input('\nWhat is the destination username?\nEnter here --> ' )

destin_host = input('\nWhat is the destination hostname?\n(eg. example.website.com)\nEnter here --> ')

print('\nLet\'s check if we have all the correct information',
'\nYour file of interest: ', fi_of_interest,
'\nYour destinations username: ', usern,
'\nYour destination hostname: ', destin_host,
        )


check = input('\nIs this correct? (y/n): ')
check = str(check)

scp_protocol = 'ssh ' + str(usern) + '@' + str(destin_host) + ':' + fi_of_interest + ' .'

if check == 'n':
    print('\nTry again...\n...Program Exiting')
    exit()
elif check == 'y':
    print('\n\nCopy the following and paste it into your terminal')
    print(' ')
    print(str(scp_protocol))
    print(' ')
else:
    print('Error 1: Closing...')
    print('maybe wrong letter or use lower case?')
    exit()

#
